AraEvent/AraEventCalibrator.cxx

```diff
@@ -871,18 +901,89 @@ void AraEventCalibrator::calibrateEvent(UsefulAtriStationEvent *theEvent, AraCal
+        hasTrimFirstBlk = TrimFirstBlock(theEvent, voltMapIt, timeMapIt, sampleList, capArrayList);
+    }    
+
+    // 5th step. Trims samples using fAtriSampleIndex table
+    // A trimmed sampleList will be used for calling the right pedestal subtraction samples and the conversion factor samples
+    if(AraCalType::hasBinWidthCalib(calType)){ 
+        BinWidthCalib(theEvent, voltMapIt, timeMapIt, sampleList, capArrayList, hasTrimFirstBlk);    
+    }
+    
+    // 6th step. Pedestal subtraction
+    if(hasPedestalSubtraction(calType)) {
+        PedSubtraction(theEvent, voltMapIt, sampleList);
+    }
+    
+    // 7th step. Common mode
+    if(hasCommonMode(calType)) {
+        CommonMode(theEvent, voltMapIt);
+    }
+
+    // 8th step. Inverts only RF channels = 0,4,8 in A3
+    // Apply conditioner function here for correct VoltCal() performance
+    if (hasInvertA3Chans(calType) && thisStationId ==3) {
+        InvertA3Chans(theEvent, voltMapIt, thisStationId);
+    }
+
+    // 9th step. Zeroing ADC WF by subtracting mean
+    // RJN change 13-Feb-2013
+    // Now do zero mean,   
+    // THM added 12-Feb-2014
+    // After zeroMean (and only then!!) do voltage calibration,
+    // KAH added 09-15-2020
+    // first, need to do voltage calibration. if you zero-mean the block beforehand, the ADC values get screwed up and the resulting waveform looks bad.
+    // each sample needs to be individually calibrated. This requires knowing the dda, the channel, the block and the sample%64. 
+    // The below code loops over each of these.
+    // the previous version of the code did the zero mean before the votlage calibration. If there are issues with A2 or A3 calibration, try switching the order back.
+    // MK added 27-09-2021
+    // Adds back just for A2 and A3. Dealing with outlier events
+    if(hasADCZeroMean(calType) && thisStationId != 5) {
+        ZeroMean(theEvent, voltMapIt);
+    }
+
+    // 10th step. Voltage calibration
+    if(hasVoltCal(calType)) {
+        VoltCal(theEvent, voltMapIt, sampleList, thisStationId);
+    }
+   
+    // 11th step. Zeroing voltage WF by subtracting mean
+    if(hasVoltZeroMean(calType)) {
+        ZeroMean(theEvent, voltMapIt);
+    }
+
+    // 12th step. Remove knwon cable delay
+    // jpd change 25-03-13
+    // now subtract off the cable delays
+    if(hasCableDelays(calType)){
+        CableDelay(theEvent, timeMapIt, unixtime, thisStationId);
+    }
+    delete sampleList, capArrayList; // delete the pointer
+
+    // fprintf(stderr, "AraEventCalibrator::CalibrateEvent() -- finished calibrating event\n");//DEBUG                        
+}
+
+void AraEventCalibrator::DAQtoEleCh(UsefulAtriStationEvent *theEvent, std::map< Int_t, std::vector <Double_t> >::iterator &voltMapIt, std::map< Int_t, std::vector <Double_t> >::iterator &timeMapIt, std::vector<std::vector<int> > *sampleList, std::vector<std::vector<int> > *capArrayList)
+{ // Converts DAQ data format to Electronic channel format
```

Comment:
For all functions please add more thorough documentation, including appropriate doxygen formatting. You can see an example on the Ray Trace Correlator: https://github.com/ara-software/AraRoot/blob/master/AraCorrelator/RayTraceCorrelator.h#L59

Reply:
Also, consider giving a more descriptive name? What is `DAQtoEleCh`? You could instead do something like `UnpackDAQFormatToElecChanFormat` or something.

Reply:
I agree the name is too simple. let me work on it. Thank you!

Reply:
These look a lot better, thanks.